CONTRIBUTING.md

```diff
@@ -125,6 +125,11 @@ differentiating between a given function and other functions, so putting them
 first makes it easier to infer the context/meaning of the function (compared to
 starting with a number of generic context-like types).
 
+#### Use `impl` where possible
+
+We prefer to use `impl ...` for arguments where possible. Generic type argument
```

Comment:
The title is "Use `impl` where possible" but the contents then restrict this to "for arguments". Should this also say something about the use of return position impl types?

Maybe also explain "where possible" a bit, this only works when there's a single use of the type.

Reply:
I added a mention of return types and tried to expand on "where possible". I'm not sure I've done this one justice (probably owing in part to not being super confident in this area myself).

Reply:
Seems pretty good to me.